### PR TITLE
docs: user-guide: add section about CRLF issues on Windows

### PR DESCRIPTION
**Pull request by efiop**
Noticed that we don't have this in the doc.

### CODE REVIEW

File: content/docs/user-guide/how-to/run-dvc-on-windows.md [hunk 1]
```diff
@@ -40,6 +40,50 @@ alternatives:
   [virtual machine](https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/),
   or in an HD partition (dual boot).
 
+## Line endings
+
+If you are using Windows and you are working with people or environments (e.g.
+production) that are not, you’ll probably run into line-ending issues at some
+point. This is because Windows uses both a carriage-return character and a
+linefeed character for newlines in its files, whereas macOS and Linux systems
+use only the linefeed character.
```
Comment (dberenbaum) [inline content/docs/user-guide/how-to/run-dvc-on-windows.md:49]:
```suggestion
linefeed character for newlines in its files (CLRF), whereas macOS and Linux systems
use only the linefeed character (LF).
```

File: content/docs/user-guide/how-to/run-dvc-on-windows.md [hunk 2]
```diff
@@ -40,6 +40,50 @@ alternatives:
...
+
+Since DVC is using content-based checksums for your pipeline dependencies,
+depending on your GIT configuration (see
+[core.autocrlf and core.eol config options](https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration)),
+DVC might see GIT-tracked files as changed, thus triggering pipeline
```
Comment (dberenbaum) [inline content/docs/user-guide/how-to/run-dvc-on-windows.md:54]:
```suggestion
depending on your Git configuration (see
[core.autocrlf and core.eol config options](https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration)),
DVC might see Git-tracked files as changed, thus triggering pipeline
```

File: content/docs/user-guide/how-to/run-dvc-on-windows.md [hunk 3]
```diff
@@ -40,6 +40,50 @@ alternatives:
...
+reproduction on `dvc repro` on one system and not on another. Thus we strongly
+recommend sticking with LF line endings when doing cross-platform work.
+
+### Configure your editor to use LF line endings
+
+Many editors on Windows will use CRLF line endings by default or even replace
+existing LF with CRLF. It is recommended that you configure your editor to stick
+to always stick to LF line endings.
```
Comment (dberenbaum) [inline content/docs/user-guide/how-to/run-dvc-on-windows.md:62]:
```suggestion
existing LF with CRLF. It is recommended that you configure your editor 
to always stick to LF line endings.
```

File: content/docs/user-guide/how-to/run-dvc-on-windows.md [hunk 4]
```diff
@@ -40,6 +40,50 @@ alternatives:
...
+### Use pre-commit hook to check and fix line endings
+
+Add this to your `.pre-commit-config.yaml` hooks:
+
+```
+      - id: mixed-line-endings
+        args: [--fix=lf]
+```
+
+to make `pre-commit` check and automatically replace all line endings with LF.
```
[the remark this answers is not in the document]
Comment (efiop) [reply content/docs/user-guide/how-to/run-dvc-on-windows.md:85]: You mean that it won't affect dvc retrospectively? Feels a bit redundant, I would assume people understand that from the `content` explanation above.

File: content/docs/user-guide/how-to/run-dvc-on-windows.md [hunk 5]
```diff
@@ -40,6 +40,50 @@ alternatives:
   [virtual machine](https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/),
   or in an HD partition (dual boot).
 
+## Line endings
+
+If you are using Windows and you are working with people or environments (e.g.
+production) that are not, you’ll probably run into line-ending issues at some
+point. This is because Windows uses both a carriage-return character and a
+linefeed character for newlines in its files (CLRF), whereas macOS and Linux systems
```
Comment (efiop) [inline content/docs/user-guide/how-to/run-dvc-on-windows.md:48]:
```suggestion
linefeed character for newlines in its files (CRLF), whereas macOS and Linux systems
```

File: content/docs/user-guide/how-to/run-dvc-on-windows.md (hunk 2 above)
Comment (efiop) [reply content/docs/user-guide/how-to/run-dvc-on-windows.md]: I'm constantly confused by the convention. I thought it should be the same as DVC

**Review by dberenbaum — APPROVED** [review]

**Pull request merged**